 vold: add in BOARD_REQUIRES_FORCE_VPARTITION

This will force mount for the virtual parition in the HP touchpad
Not sure if any other board has this issue but here is a way
around it

diff --git a/Android.mk b/Android.mk
index 00550c3..6a68701 100644
--- a/Android.mk
+++ b/Android.mk
@@ -96,6 +96,10 @@ ifeq ($(TARGET_KERNEL_HAVE_NTFS),true)
 vold_cflags += -DCONFIG_KERNEL_HAVE_NTFS
 endif
 
+ifeq ($(BOARD_REQUIRES_FORCE_VPARTITION),true)
+vold_cflags += -DCONFIG_FORCE_VPARTITION
+endif
+
 include $(CLEAR_VARS)
 
 LOCAL_ADDITIONAL_DEPENDENCIES := $(LOCAL_PATH)/Android.mk
diff --git a/Disk.cpp b/Disk.cpp
index 00550c3..6a68711 100644
--- a/Disk.cpp
+++ b/Disk.cpp
@@ -107,8 +107,12 @@ static bool isVirtioBlkDevice(unsigned int major) {
      * "ranchu", the device's sysfs path should end with "/block/vd[d-z]", etc.
      * But just having a) and b) is enough for now.
      */
+#ifdef CONFIG_FORCE_VPARTITION
+    return true;
+#else
     return IsRunningInEmulator() && major >= kMajorBlockExperimentalMin
             && major <= kMajorBlockExperimentalMax;
+#endif
 }
 
 Disk::Disk(const std::string& eventPath, dev_t device,
